presto-main/src/main/java/com/facebook/presto/cost/JoinStatsRule.java

```diff
@@ -158,9 +161,14 @@ private PlanNodeStatsEstimate computeInnerJoinStats(JoinNode node, PlanNodeStats
         }
 
         PlanNodeStatsEstimate equiJoinEstimate = filterByEquiJoinClauses(crossJoinStats, node.getCriteria(), session, types);
-
         if (equiJoinEstimate.isOutputRowCountUnknown()) {
-            return PlanNodeStatsEstimate.unknown();
+            double defaultJoinSelectivityFactor = getDefaultJoinSelectivityCoefficient(session);
+            if (defaultJoinSelectivityFactor == DEFAULT_JOIN_SELECTIVITY_DISABLED) {
+                return PlanNodeStatsEstimate.unknown();
+            }
+            else {
+                equiJoinEstimate = crossJoinStats.mapOutputRowCount(joinSourceRowCount -> crossJoinStats.getOutputRowCount() * defaultJoinSelectivityFactor);
```

Comment:
If one (or both) of the input join cardinalities is unknown (e.g. aggregation stats are unknown), will we get unknown as the join cardinality? (I think so but please confirm.)

Reply:
yes if any side of join leg has an unknown cardinality, the join stats will be unknown.

Reply:
actually for the outer join case, it's safer to check the joinComplementStats and if it's unknown stats, we will just return inner join stats.